cpp/include/cudf/detail/scatter.hpp

```diff
@@ -144,6 +144,35 @@ std::vector<std::unique_ptr<table>> scatter_to_tables(
     rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
     cudaStream_t stream = 0);
 
+
+/**
+ * @copydoc cudf::experimental::boolean_mask_scatter(
+ *                    column_view const& source, column_view const& target,
+ *                    column_view const& boolean_mask,
+ *                    rmm::mr::device_memory_resource *mr)
+ *
+ * @param stream The stream to use for CUDA operations
+ */
+std::unique_ptr<column> boolean_mask_scatter(
```

Comment:
The existing scatter function works on tables, not columns. Is there a reason to make the new API inconsistent in this regard?

Reply:
Here you will be having dense buffer or a scalar as input, so the possibility of having a table would be less as type should match. But this can be extended to a table scenario if we can have vector of scalars/dense buffers for each column.

Reply:
I don't understand how this API is different from scatter, even the implementation calls it internally. Types need to match in either case. The only difference is that you need to compute a scatter map from the bool_mask. 
Is the expected usage different?

Reply:
I think it is better to be in sync with scatter, so I will update the API to handle the table.